code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -368,12 +370,20 @@
 				success = TRUE
 
 		if(success)
-			F = F.make_plating(TRUE) || F
-			if(was_replacing && tilestack)
+			var/area/is_this_maints = get_area(F)
+			if(was_replacing && tilestack)	//turn the tile into plating (if needed), then replace it
+				F = F.make_plating(TRUE) || F
 				tilestack.place_tile(F)
 				if(!tilestack)
 					speak("Requesting refill of custom floor tiles to continue replacing.")
-			else
+			else if(F.broken || F.burnt)	//repair the tile and reset it to be undamaged (rather than replacing it)
+				F.broken = FALSE
+				F.burnt = FALSE
+				F.icon_state = initial(F.icon_state)
+			else if(findtext(is_this_maints.name, "maint"))	//place catwalk if it's plating and we're in maints
```

Comment:
This is not an acceptable way to detect maintenance. I think there's flags you can use?

Reply:
I code dived and the only two area helper procs are `isarea()` and `get_area()` both of which cannot be used to check if a turf is maint.

Then I looked at the flags as an option:

https://github.com/tgstation/tgstation/blob/a3a174f031ecd81773f55bffef1f8f152fddada2/code/game/area/space_station_13_areas.dm#L168-L173

https://github.com/tgstation/tgstation/blob/11302feef45d2bc0fc717ee426114c2c7e52caeb/code/_globalvars/bitfields.dm#L52-L70

It doesn't appear there is a maintenance flag set anywhere for the codebase.  Looks like a new one will need to be made.  I wonder if others used similar hacks like this to check if a turf was maintenance...

Reply:
this was the method used for floorbots' `is_hull_breach()` proc, i assumed it'd be fine

`/mob/living/simple_animal/bot/floorbot/proc/is_hull_breach(turf/t) //Ignore space tiles not considered part of a structure, also ignores shuttle docking areas.`
`>>>var/area/t_area = get_area(t)`
`>>>if(t_area && (t_area.name == "Space" || findtext(t_area.name, "huttle")))`
`(more code past here`

Reply:
as for the multi-z lighting thing, it most likely doesnt fix nor damage that; these are all just reskins of the original catwalk

Reply:
> I code dived and the only two area helper procs are `isarea()` and `get_area()` both of which cannot be used to check if a turf is maint.
> 
> Then I looked at the flags as an option:

Sorry should have been more clear. They should add a bitflag that's "is maintenance" rather then searching the name, because names can change and that's not a stable way to check

I have no doubt boomer code did this too, but I dream of a better tomorrow

Reply:
I can tryyyy to fix this? Maybe? Lemme look into it.

Reply:
added, called "IS_MAINTINENCE_AREA"

Reply:
marking this as resolved, if i did something wrong please re-review hnnnnhbhhgbg